### PR TITLE
Added watch.sh

### PR DESCRIPTION
**Pull request by johannesloetzsch**
Use inotifywait (from inotify-tools) to rebuild everytime the SOURCE changed

### CODE REVIEW

**Comment by johannesloetzsch** [comment]
Till now, there is one minor issue:
In case of syntax errors, `doclog.sh` keeps running, prompting the user in a prolog repl.
When using `watch.sh` this is not intended. It would be nice to exit doclog after errors instead.

I could work around this in `watch.sh` but I would prefer, to have this as default behavior of `doclog.sh`. What do you think @aarroyoc ?

**Comment by aarroyoc** [comment]
It should be possible to add a `catch/3` that prevents going back to the REPL. Give me a few days

**Comment by johannesloetzsch** [comment]
> It should be possible to add a `catch/3` that prevents going back to the REPL. Give me a few days

Thank you :)

Even without I, used DocLog with watch.sh quite extensively and successful :)
Till now when errors occur. Ctrl+d closes the repl and continues the watch loop.

I wrote a little tutorial about how to use prolog + scryer + playground. I assume I will push it soon and will let you know than… 

Are you interested in a listing of issues, workarounds and error messages I experienced, while working with DocLog?

**Comment by triska** [comment]
Previous related discussion: https://github.com/mthom/scryer-prolog/issues/187#issuecomment-621915840

**Comment by aarroyoc** [comment]
While we implement that on Scryer, I pushed a commit that wraps everything in a `catch/3`. Let me know if it works for you

@triska maybe that flag deserves its own issue as the original one was closed

**Comment by triska** [comment]
> maybe that flag deserves its own issue as the original one was closed

Yes thank you, good point, I have filed https://github.com/mthom/scryer-prolog/issues/2651 for it!

**Comment by triska** [comment]
As mentioned by @hurufu in https://github.com/mthom/scryer-prolog/issues/2651#issuecomment-2450869081, multiple `-g` switches can be used, so adding `-g halt` to the invocation may suffice to solve this more elegantly!

**Comment by aarroyoc** [comment]
Does the proposed solution by @triska work for you?

**Comment by johannesloetzsch** [comment]
> Does the proposed solution by @triska work for you?

Yes, seems to work well :)

From my side, this PR is ready to be merged

File: doclog.sh [hunk 1]
```diff
@@ -5,5 +5,5 @@ SOURCE=$(realpath $1)
 OUTPUT=$(realpath $2)
 
 cd $DOCLOG
-scryer-prolog -g 'run("'$SOURCE'", "'$OUTPUT'").' main.pl
+scryer-prolog -g 'run("'$SOURCE'", "'$OUTPUT'").' -g 'halt.' main.pl
```
Comment (triska) [inline doclog.sh:8]: `-g halt` should work.

File: doclog.sh [hunk 2]
```diff
@@ -5,5 +5,5 @@ SOURCE=$(realpath $1)
 OUTPUT=$(realpath $2)
 
 cd $DOCLOG
-scryer-prolog -g 'run("'$SOURCE'", "'$OUTPUT'").' main.pl
+scryer-prolog -g 'run("'$SOURCE'", "'$OUTPUT'").' -g 'halt' main.pl
```
Comment (triska) [inline doclog.sh:8]: As mentioned, I think `-g halt` would work.

**Pull request merged**